src/docfx/build/toc/parser/TableOfContentsParser.cs

```diff
@@ -88,25 +90,31 @@ private static (List<Error>, TableOfContentsModel) LoadTocModel(JToken tocToken)
             Document rootPath,
             List<Document> referencedFiles,
             List<Document> referencedTocs,
-            List<Document> parents,
             string content = null)
         {
             // add to parent path
-            if (parents.Contains(file))
+            if (t_recursionDetector.Value.Contains(file))
```

Comment:
why do we change toc parts?

Reply:
Trying to use the same way to detect recursion